File: dotnet/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/GUIFactoryProvider.cs
namespace AbstractFactoryDesignPattern.Example1;

public static class GUIFactoryProvider
{
	public static GUIFactory GetFactory(string platform)
	{
		return platform.ToLower() switch
		{
			"windows" => new WindowsFactory(),
			"mac" => new MacFactory(),
			"linux" => new LinuxFactory(),
			_ => throw new ArgumentException("Invalid OS type", nameof(platform))
		};
	}
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/AbstractFactoryDesignPattern/AbstractFactoryDesignPattern.Example1/src/GUIFactoryProvider.ts
import { GUIFactory } from './factory/abstract/GUIFactory';
import { WindowsFactory } from './factory/concrete/WindowsFactory';
import { MacFactory } from './factory/concrete/MacFactory';
import { LinuxFactory } from './factory/concrete/LinuxFactory';

export class GUIFactoryProvider {
  static getFactory(platform: string): GUIFactory {
    switch (platform.toLowerCase()) {
      case 'windows': return new WindowsFactory();
      case 'mac': return new MacFactory();
      case 'linux': return new LinuxFactory();
      default: throw new Error(`Invalid platform: ${platform}`);
    }
  }
}
